agent/engine/task_manager.go

```diff
@@ -985,6 +983,9 @@ func (mtask *managedTask) progressTask() {
 	// these and if we exit early transitions can exit the goroutine and it'll
 	// get GC'd eventually
 	resources := mtask.GetResources()
+	for _, container := range mtask.Containers {
+		resources = append(resources, container.GetResources()...)
```

Comment:
it doesn't look like we need to create a new slice here. We just need the length.
Can we change this to:

```golang
resourceLen := len(mtask.GetResources())
for _, container := range mtask.Containers {
    resourceLen += len(container.GetResources())
}
transitionChange := make(chan struct{}, len(mtask.Containers)+resourceLen)
transitionChangeEntity := make(chan string, len(mtask.Containers)+resourceLen)
```

Reply:
haha right 😄